pkg/thing/delivery/http/thing_proxy.go

```diff
@@ -179,7 +179,7 @@ func (p proxy) Get(authorization, ID string) (*entities.Thing, error) {
 		}
 	}
 
-	return nil, &entities.ErrThingNotFound{ID: ID}
+	return nil, entities.ErrThingNotFound
```

Comment:
There is one thing we need to be aware of before changing the error pattern, some of them were carrying contextual information about the error like this one (which was informing that a specific device couldn't be found). How can we add this information? It was because of these situations that you suggested helper functions to create them, something link `NewErrThingNotFound(id)`?

Reply:
@joaoaneto Since the ID is in the calling context, I think there is no problem to ommit the ID in the returned error. Who gets the error will know waht ID is relationed to error.

I suggested helper funcions to help adding contextual info to errors, for example where we are using `fmt.Errorf` to wrap two errors, we could create a function that nest this wrap for us.